resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/k8s/KubernetesExecutorBackend.scala

```diff
@@ -66,7 +66,8 @@ private[spark] object KubernetesExecutorBackend extends Logging {
 
     SparkHadoopUtil.get.runAsSparkUser { () =>
       // Debug code
-      Utils.checkHost(arguments.hostname)
+      assert(arguments.hostname != null &&
+          (arguments.hostname.indexOf(':') == -1 || arguments.hostname.split(":").length > 2))
```

Comment:
why not fix `Utils.checkHost`?

Reply:
The original code of `Utils.checkHost` is made in YARN age.
And, that is a valid check in YARN environment's URL.
Please see the link in the PR description.

<img width="460" alt="Screen Shot 2022-06-09 at 9 11 24 PM" src="https://user-images.githubusercontent.com/9700541/172988832-03ccea4e-0f12-4093-a59c-045eec86292c.png">

Reply:
We don't know how YARN or other resource managers are configured with IPv6 in these days.